db/db.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return 0, err
+	}
+	id, err := res.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	return id, nil
+}
+
+// Структура задачи
+type Task struct {
+	ID      string `json:"id"`
+	Date    string `json:"date"`
+	Title   string `json:"title"`
+	Comment string `json:"comment"`
+	Repeat  string `json:"repeat"`
+}
+
+// Возвращаем список ближайших задач из базы данных
+// В задании этого нет, но если фронтенд будет поддерживать пагинацию, то это пригодится
+
+func GetTasks(limit, offset int) ([]Task, error) {
+	now := time.Now().Format("20060102")
+	query := `SELECT id, date, title, comment, repeat FROM scheduler WHERE date >= ? ORDER BY date LIMIT ? OFFSET ?`
+	rows, err := DB.Query(query, now, limit, offset)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var tasks []Task
+	for rows.Next() {
+		var task Task
+		var id int64
+		err := rows.Scan(&id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			return nil, err
+		}
+		task.ID = fmt.Sprintf("%d", id)
+		tasks = append(tasks, task)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+// Возвращаем задачи по заданной дате
+func GetTasksByDate(date string, limit, offset int) ([]Task, error) {
+	query := `SELECT id, date, title, comment, repeat FROM scheduler WHERE date = ? ORDER BY date LIMIT ? OFFSET ?`
+	rows, err := DB.Query(query, date, limit, offset)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var tasks []Task
+	for rows.Next() {
+		var task Task
+		var id int64
+		err := rows.Scan(&id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			return nil, err
+		}
+		task.ID = fmt.Sprintf("%d", id)
+		tasks = append(tasks, task)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+// Выполняем поиск задач по подстроке в заголовке или комментарии
+func SearchTasks(search string, limit, offset int) ([]Task, error) {
+	searchTerm := "%" + search + "%"
+	query := `SELECT id, date, title, comment, repeat FROM scheduler WHERE title LIKE ? OR comment LIKE ? ORDER BY date LIMIT ? OFFSET ?`
+	rows, err := DB.Query(query, searchTerm, searchTerm, limit, offset)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var tasks []Task
+	for rows.Next() {
+		var task Task
+		var id int64
+		err := rows.Scan(&id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			return nil, err
+		}
+		task.ID = fmt.Sprintf("%d", id)
+		tasks = append(tasks, task)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+// Ввозвращаем задачу по её идентификатору
+func GetTaskByID(id int64) (Task, error) {
+	query := `SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?`
+	row := DB.QueryRow(query, id)
+
+	var task Task
+	var taskID int64
+	err := row.Scan(&taskID, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Для сравнивания ошибок нужно использовать специальные функции
```suggestion
		if errors.Is(err,sql.ErrNoRows) {
```

Reply:
fxd, спасибо!